core/src/main/java/io/grpc/internal/DnsNameResolver.java

```diff
@@ -200,17 +217,33 @@ public void run() {
           }
 
           ResolutionResults resolutionResults;
-          try {
-            ResourceResolver resourceResolver = null;
-            if (enableJndi) {
-              resourceResolver = getResourceResolver();
+          long networkAddressCacheTtl = getNetworkAddressCacheTtl();
+          boolean refreshRequired = networkAddressCacheTtl <= 0
```

Comment:
This should be named resourceRefreshRequired, to distinguish it from the address results needing refresh.